lib/api/v3/time_entries/time_entries_api.rb

```diff
@@ -105,6 +105,14 @@ class TimeEntriesAPI < ::API::OpenProjectAPI
                 fail ::API::Errors::ErrorBase.create_and_merge_errors(result.errors)
               end
             end
+
+            delete do
+              if ::TimeEntries::DeleteService.new(time_entry: @time_entry, user: current_user).call
+                status 202
+              else
+                fail ::API::Errors::Unauthorized
+              end
+            end
```

Comment:
Once the service returns a service result (please see above), this should be changed to:

```
call = ::TimeEntries::DeleteService.new(time_entry: @time_entry, user: current_user).call

if call.success?
  status 202
else
  fail ::API::Errors::ErrorBase.create_and_merge_errors(call.errors)
end
```

That way, the api controller does not need to know why the service call failed, it just knows that it did, and will return an appropriate error based on the errors provided by the service.